keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -64,26 +97,28 @@ class SqlArtifactRepository(
 
   override fun getAll(type: ArtifactType?): List<DeliveryArtifact> =
     jooq
-      .select(DELIVERY_ARTIFACT.NAME, DELIVERY_ARTIFACT.TYPE)
+      .select(DELIVERY_ARTIFACT.NAME, DELIVERY_ARTIFACT.TYPE, DELIVERY_ARTIFACT.DETAILS)
       .from(DELIVERY_ARTIFACT)
       .apply { if (type != null) where(DELIVERY_ARTIFACT.TYPE.eq(type.toString())) }
-      .fetch { (name, type) ->
-        DeliveryArtifact(name, ArtifactType.valueOf(type))
+      .fetch { (name, storedType, details) ->
+        ArtifactUtils.mapToArtifact(name, ArtifactType.valueOf(storedType), details)
       }
 
+  override fun versions(name: String, type: ArtifactType, statuses: List<ArtifactStatus>): List<String> =
+    versions(get(name, type), statuses)
+
   override fun versions(artifact: DeliveryArtifact, statuses: List<ArtifactStatus>): List<String> {
-    val status = statuses.map { it.toString() }
     return if (isRegistered(artifact.name, artifact.type)) {
       jooq
         .select(DELIVERY_ARTIFACT_VERSION.VERSION)
         .from(DELIVERY_ARTIFACT, DELIVERY_ARTIFACT_VERSION)
         .where(DELIVERY_ARTIFACT.UID.eq(DELIVERY_ARTIFACT_VERSION.DELIVERY_ARTIFACT_UID))
         .and(DELIVERY_ARTIFACT.NAME.eq(artifact.name))
         .and(DELIVERY_ARTIFACT.TYPE.eq(artifact.type.name))
-        .and(DELIVERY_ARTIFACT_VERSION.STATUS.`in`(*status.toTypedArray()))
+        .apply { if (statuses.isNotEmpty()) and(DELIVERY_ARTIFACT_VERSION.STATUS.`in`(*statuses.map { it.toString() }.toTypedArray())) }
         .fetch()
         .getValues(DELIVERY_ARTIFACT_VERSION.VERSION)
-        .sortAppVersion()
+        .sortedWith(artifact.versioningStrategy.comparator)
```

Comment:
⭐️ 🕵️‍♀️  This is the actual functionality I was going for with this PR: an artifact has sorting defined on it, and that sorting is used to correctly return the versions, and the "latest" version.